cranelift/codegen/src/machinst/lower.rs

```diff
@@ -1387,15 +1387,28 @@ impl<'func, I: VCodeInst> Lower<'func, I> {
         let inst_data = self.data(ir_inst);
         match inst_data {
             InstructionData::Shuffle { imm, .. } => {
-                let buffer = self.f.dfg.immediates.get(imm.clone()).unwrap().as_slice();
-                let value = DataValue::V128(buffer.try_into().expect("a 16-byte data buffer"));
+                let mask = self.f.dfg.immediates.get(imm.clone()).unwrap().as_slice();
+                let value = if mask.len() == 16 {
```

Comment:
Can we turn this `if`-chain into a `match` (with `16 => ..., 8 => ..., _ => panic!()`)?